src/deno_dir.rs

```diff
@@ -481,5 +481,24 @@ fn test_resolve_module() {
 const ASSET_PREFIX: &str = "/$asset$/";
 
 fn is_remote(module_name: &str) -> bool {
-  module_name.starts_with("http")
+  let url = Url::parse(module_name);
```

Comment:
Are we calling Url::parse twice now during CodeFetch ? :/

Reply:
We are calling it once on module_specifier in resolve_module, and once in this function, I think it's fine, this operation is nullipotent and both variables are string that can either be an Url or a local path.

We could change those to use Url instead, but Urls cannot be relative and atm, the code assumes they are strings. Maybe use a type that can be either an Url or a Path.

Or we can let the Url parsing here.

Reply:
@NewLunarFire I'd prefer it if we could figure out how to not parse the URL twice.